pkg/scale/varying_data_type.go

```diff
@@ -66,8 +66,11 @@ func (vdt *VaryingDataType) Set(value VaryingDataTypeValue) (err error) {
 }
 
 // Value returns value stored in vdt
-func (vdt *VaryingDataType) Value() VaryingDataTypeValue {
-	return vdt.value
+func (vdt *VaryingDataType) Value() (VaryingDataTypeValue, error) {
+	if vdt.value == nil {
+		return nil, ErrVaryingDataTypeNotSet
```

Comment:
nit to force using errors.Is instead of `==`

```suggestion
		return nil, fmt.Errorf("%w", ErrVaryingDataTypeNotSet)
```